### PR TITLE
validate trait scores

### PR DESCRIPTION
**Pull request by malaffoon**

### CODE REVIEW

File: exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/model/Assessment.java [hunk 1]
```diff
@@ -17,9 +17,9 @@
     private int subjectId;
     private String subjectCode;
     private List<AssessmentScore> scores;
+    private List<AssessmentTrait> traits;
     private Map<String, Integer> targetCodeToId;
     private Map<String, Integer> items;
-    private Map<String, Integer> traits;
```
Comment (malaffoon) [inline exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/model/Assessment.java:22]: Need more info than just the trait code and id.

File: exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/model/ExamTrait.java [hunk 2]
```diff
@@ -6,7 +6,7 @@
 public class ExamTrait {
     private Long id;
     private int traitId;
-    private int score;
+    private Integer score;
```
Comment (malaffoon) [inline exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/model/ExamTrait.java:9]: The way parsing and validating works, we have to allow a null value to be set, even though that is logically invalid.

File: exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultStudentExamProcessor.java [hunk 3]
```diff
@@ -1,54 +1,55 @@
 package org.opentestsystem.rdw.ingest.processor.service.impl;
 
 import com.google.common.collect.ImmutableList;
-import org.opentestsystem.rdw.ingest.processor.model.ExamTrait;
```
Comment (malaffoon) [inline exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultStudentExamProcessor.java:4]: Hmm, i did "optimize imports" but import settings must be different than the old days.

File: exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultStudentExamProcessor.java [hunk 4]
```diff
@@ -313,4 +314,17 @@ private Integer toValidCompleteness(final String value) {
 
         throw new IllegalArgumentException("invalid value, must be a positive number");
     };
+
+    private static Function<String, Integer> traitScoreChecker(final Integer maxScore) {
```
Comment (malaffoon) [inline exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultStudentExamProcessor.java:318]: The point of the PR: trait scores must be integers between 0 and the max score. If the max score isn't defined, then  just check that it is a non-negative integer.

Comment (gregcharles) [reply exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultStudentExamProcessor.java:318]: 👍

**Pull request merged**